### PR TITLE
Upgrade the versions of Android support libraries

### PR DESCRIPTION
**Pull request by felipeerias**
This PR upgrades the versions of Android support libraries used by Wolvic.

First, we upgrade the Kotlin version to 1.8.10 and set the project's `compileSdkVersion` to `31`, as this is required by some of the updated libraries.

Then, we update the following Android libraries as much as possible while keeping the `compileSdkVersion` at `31` for compatibility:

- androidx.annotation 1.6.0
- androidx.appcompat 1.4.2
- androidx.recyclerview 1.2.1
- androidx.cardview 1.0.0
- android.material 1.6.1
- androidx.vectordrawable 1.1.0
- androidx.viewpager2 1.0.0

Finally, we fix some UI issues caused by the upgraded libraries:

- App theme "FxR.Dark" extends "Theme.Material3.Dark.NoActionBar"
- AppCompatTextView replaced by MaterialTextView.
- UITextButton extends MaterialButton instead of AppCompatButton.
- Fix small UI issues at setting_switch.xml and navigation_url.xml.
- Use app:tint instead of android:tint.

### CODE REVIEW

File: versions.gradle [hunk 1]
```diff
@@ -185,6 +185,7 @@ ext.deps = deps
 def build_versions = [:]
 build_versions.min_sdk = 24
 build_versions.min_sdk_wave = 25
+build_versions.compile_sdk = 31
```
Comment (svillar) [inline versions.gradle:186]: I wonder, is this required by any of the updated dependencies?

File: app/src/common/shared/com/igalia/wolvic/ui/views/UITextButton.java [hunk 2]
```diff
@@ -12,11 +12,11 @@
 import android.util.AttributeSet;
 
 import androidx.annotation.IdRes;
-import androidx.appcompat.widget.AppCompatButton;
 
+import com.google.android.material.button.MaterialButton;
 import com.igalia.wolvic.R;
 
-public class UITextButton extends AppCompatButton implements CustomUIButton {
+public class UITextButton extends MaterialButton implements CustomUIButton {
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/views/UITextButton.java:19]: This is what we need! I was getting tons of errors related to this when porting a couple of new architectures we're going to support.

File: app/build.gradle [hunk 3]
```diff
@@ -103,7 +103,7 @@ ext.gleanGenerateMarkdownDocs = true
 ext.gleanDocsDirectory = "$rootDir/docs"
 
 android {
-    compileSdkVersion build_versions.target_sdk
+    compileSdkVersion build_versions.compile_sdk
```
Comment (svillar) [inline app/build.gradle:106]: Shouldn't this be part of `defaultConfig`?

**Comment by svillar** [comment]
Seems like this breaks all the annotations. `Cannot resolve symbol Nullable` or `Cannot resolve symbol Keep` all around

**Comment by svillar** [comment]
There are other issues. Apparently some of the dependencies force the `minSdkVersion` to 31. We cannot make that change for some of the platforms we (will) support.

File: versions.gradle (hunk 1 above)
Comment (felipeerias) [reply versions.gradle:186]: Yes, it is required by `androidx.appcompat:appcompat`:

- version **1.4.2** requires **31**
- version **1.5.1** requires **32** 
- version **1.6.1** (the latest) requires **33**

File: app/build.gradle (hunk 3 above)
Comment (felipeerias) [reply app/build.gradle:106]: Done.

File: versions.gradle (hunk 1 above)
Comment (felipeerias) [reply versions.gradle:186]: We will use `androidx.appcompat:appcompat` **1.4.2** and `compileSdkVersion` **31**.

File: app/build.gradle [hunk 4]
```diff
@@ -103,10 +103,10 @@ ext.gleanGenerateMarkdownDocs = true
 ext.gleanDocsDirectory = "$rootDir/docs"
 
 android {
-    compileSdkVersion build_versions.compile_sdk
     defaultConfig {
         applicationId "com.igalia.wolvic"
         minSdkVersion build_versions.min_sdk
+        compileSdkVersion build_versions.compile_sdk
```
Comment (svillar) [inline app/build.gradle:109]: Sorry I was wrong, the `compilesdk` should be placed just under `android`

**Review by svillar — CHANGES_REQUESTED** [review]

File: versions.gradle (hunk 1 above)
Comment (svillar) [reply versions.gradle:186]: Nit: this is a bit stupid but mind moving the compilesdk *after* the target ones? I think it visually makes explicity that mindsdk<=targetsdk<=compilesdk

**Review by svillar — CHANGES_REQUESTED** [review]

**Review by svillar — APPROVED** [review]
Nice job. Let's modernize Wolvic

**Pull request merged**